exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/DiagnosisKeyDao.java

```diff
@@ -54,8 +54,8 @@ public int deleteVerificationsBefore(Instant verificationTime) {
     }
 
     @Transactional
-    public void addKeys(int verificationId, String requestChecksum, int interval, List<TemporaryExposureKey> keys) {
-        if (verify(verificationId, requestChecksum) && !keys.isEmpty()) {
+    public void addKeys(int verificationId, String requestChecksum, int interval, List<TemporaryExposureKey> keys, long exportedKeyCount) {
```

Comment:
The filtering is in the DAO, so I think the exportedCount calculation might as well be also. That would avoid making so many test-changes since you wouldn't need to change the function args.